src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -1166,7 +1166,7 @@ internal ImageClassificationEstimator(IHostEnvironment env, Options options, Dnn
             _options = options;
             _dnnModel = dnnModel;
             _tfInputTypes = new[] { TF_DataType.TF_STRING };
-            _outputTypes = new[] { new VectorDataViewType(NumberDataViewType.Single), NumberDataViewType.UInt32.GetItemType() };
+            _outputTypes = new DataViewType[] { new VectorDataViewType(NumberDataViewType.Single), new KeyDataViewType(typeof(uint), 5) };
```

Comment:
>5 [](start = 133, length = 1)

You are hard coding the number of classes as 5, this will vary from scenario to scenario.